src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -265,11 +266,21 @@ public void discardStagedChanges() {
 
     @Override
     public List<HistoryRecord> revertTo(HistoryRecord record) throws NoSuchElementException {
-        List<HistoryRecord> poppedRecords = historyManager.popRecordsTo(record);
+        List<HistoryRecord> poppedRecords = historyManager.popRecordsTo(record, stagedAddressBook);
         changeBaseTo(record.getCopyOfAddressBook());
         return poppedRecords;
     }
 
+    @Override
+    public HistoryRecord redo() throws IllegalStateException {
+        Optional<HistoryRecord> redoneRecord = historyManager.popRedo(stagedAddressBook);
+        if (redoneRecord.isEmpty()) {
+            throw new IllegalStateException("Cannot redo: previous command was not an undo");
```

Comment:
Previous mutator command? Can I `redo` if I `undo` then use a non mutator command?

Reply:
Yes, non `MutatorCommand`s will not affect undo/redo/history at all.